sphinx_gallery/sorting.py

```diff
@@ -40,18 +40,43 @@ def __init__(self, ordered_list):
                 "the paths of each gallery subfolder"
             )
 
-        self.ordered_list = list(os.path.normpath(path) for path in ordered_list)
+        self.ordered_list = [
+            "*" if path == "*" else os.path.normpath(path) for path in ordered_list
+        ]
+        try:
+            i = ordered_list.index("*")
+            self.has_wildcard = True
+            self.ordered_list_start = self.ordered_list[:i]
+            self.ordered_list_end = self.ordered_list[i + 1 :]
+        except ValueError:  # from index("*")
+            self.has_wildcard = False
+            self.ordered_list_start = []
+            self.ordered_list_end = self.ordered_list
 
     def __call__(self, item):
-        """Return index of item in `ordered_list`, raising error if it is missing."""
-        if item in self.ordered_list:
-            return self.ordered_list.index(item)
+        """
+        Return an integer suited for ordering the items.
+
+        If the ordered_list contains a wildcard "*", items before "*" will return
+        negative numbers, items after "*" will have positive numbers, and
+        not-listed items will return 0.
+
+        If there is no wildcard, all items with return positive numbers, and
+        not-listed items will raise a ConfigError.
+        """
+        if item in self.ordered_list_start:
+            return self.ordered_list_start.index(item) - len(self.ordered_list_start)
+        elif item in self.ordered_list_end:
+            return self.ordered_list_end.index(item) + 1
         else:
-            raise ConfigError(
-                "If you use an explicit folder ordering, you "
-                "must specify all folders. Explicit order not "
-                "found for {}".format(item)
-            )
+            if self.has_wildcard:
+                return 0
+            else:
+                raise ConfigError(
+                    "If you use an explicit folder ordering, you "
+                    "must specify all folders. Explicit order not "
+                    "found for {}".format(item)
```

Comment:
I've reworded the whole message, putting the problem first, and explanation/solution second, including '*'. Also focused on 'subsection_order' config, which is the context from which the error raises and where it needs to be fixed. I've left out "see docs", because I find such generic references not too helpful.